File: crates/prolly/bindings/ruby/examples/cookbook_scenarios.rb
# frozen_string_literal: true

require 'rbconfig'

SCENARIOS = [
  'batch_build.rb',
  'local_first_state.rb',
  'resolver.rb',
  'crdt_merge.rb',
  'conversation_memory.rb',
  'agent_event_log.rb',
  'background_compaction.rb',
  'deterministic_rag_snapshot.rb',
  'document_chunk_index.rb',
  'vector_sidecar.rb',
  'provenance_values.rb',
  'materialized_view.rb',
  'filesystem_snapshot.rb',
  'durable_sqlite.rb',
].freeze

here = __dir__
SCENARIOS.each do |scenario|
  ok = system(RbConfig.ruby, File.join(here, scenario))
  exit($?.exitstatus || 1) unless ok
end

File: crates/prolly/bindings/python/examples/cookbook_scenarios.py
from __future__ import annotations

import subprocess
import sys
from pathlib import Path

SCENARIOS = [
    "batch_build.py",
    "local_first_state.py",
    "resolver.py",
    "crdt_merge.py",
    "conversation_memory.py",
    "agent_event_log.py",
    "background_compaction.py",
    "deterministic_rag_snapshot.py",
    "document_chunk_index.py",
    "vector_sidecar.py",
    "provenance_values.py",
    "materialized_view.py",
    "filesystem_snapshot.py",
    "durable_sqlite.py",
]


def main() -> None:
    here = Path(__file__).resolve().parent
    for scenario in SCENARIOS:
        subprocess.run([sys.executable, str(here / scenario)], check=True)


if __name__ == "__main__":
    main()
